server/src/main/java/org/elasticsearch/transport/OutboundHandler.java

```diff
@@ -237,7 +236,7 @@ private void internalSend(
         final long messageSize = reference.length();
         TransportLogger.logOutboundMessage(channel, reference);
         // stash thread context so that channel event loop is not polluted by thread context
-        try (ThreadContext.StoredContext existing = threadPool.getThreadContext().stashContext()) {
+        try (var ignored = threadPool.getThreadContext().newEmptyContext()) {
```

Comment:
Same question: why this changed?

Reply:
Shouldn't really matter either way but we generally expect the thread context to be totally empty on the Netty event loop. This was one case where stashing used to do the right thing but today might leak tracing-related headers.